src/challenges.js

```diff
@@ -40,6 +56,10 @@ const checkMatchingLeaves = obj => {
 const flatten = elements => {
   // Flattens a nested array (the nesting can be to any depth).
   // Example: flatten([1, [2], [3, [[4]]]]); => [1, 2, 3, 4];
```

Comment:
This is really close but you usually want to assign reduce to it's own variable and then return it.  Otherwise good job!!!